sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -317,9 +340,8 @@ public EventHubBufferedProducerClient(string connectionString) : this(connection
         public EventHubBufferedProducerClient(string fullyQualifiedNamespace,
                                               string eventHubName,
                                               AzureNamedKeyCredential credential,
-                                              EventHubBufferedProducerClientOptions clientOptions = default) : this(clientOptions)
+                                              EventHubBufferedProducerClientOptions clientOptions = default) : this(fullyQualifiedNamespace, eventHubName, (object)credential, clientOptions)
```

Comment:
Is this cast required? 

```suggestion
                                              EventHubBufferedProducerClientOptions clientOptions = default) : this(fullyQualifiedNamespace, eventHubName, credential, clientOptions)
```

Reply:
Sadly, it was.   It kept giving me a message about trying to delegate to the same constructor.  I'm guessing the overload resolution was picking up on the stronger type match.

Reply:
We had to do the same for `EventProcessor<T>`:  https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs#L325